### PR TITLE
Update the Lite custom element parser

### PR DESCRIPTION
**Pull request by whitphx**
## Description

With this PR, using the custom component like below becomes possible where the child elements such as `<gradio-requirements>` are used with the text literal directly inside the top-level `<gradio-lite>`.
```html
<gradio-lite>
import gradio as gr

gr.Interface(
	fn=lambda x: x,
	inputs=gr.Textbox(),
	outputs=gr.Textbox()
).launch()

<gradio-requirements>
matplotlib
</gradio-requirements>
</gradio-lite>
```

Before this PR, the `<gradio-code>` element is needed to declare the source code with using `<gradio-requirements>`.

### CODE REVIEW

File: js/app/src/lite/custom-element/index.ts [hunk 1]
```diff
@@ -157,13 +152,24 @@ export function bootstrap_custom_element(
...
-				);
+			if (codeElements.length === 0) {
+				// If there is no <gradio-code> element, try to parse the content of the custom element as code.
+				let code = "";
+				this.childNodes.forEach((node) => {
+					if (node.nodeType === Node.TEXT_NODE) {
+						code += node.textContent;
+					}
+				});
+				options.code = code.trim() || undefined;
```
Comment (aliabd) [inline js/app/src/lite/custom-element/index.ts:163]: Why are you using `.trim` here? 

I think it's conflicting with my `clean_indent` function which is causing a lot of indentation issues. See the index.html file. 

https://github.com/gradio-app/gradio/blob/06bdf0eddf59bf79b6b4d1268fc9290955ef2490/js/app/src/lite/index.ts#L93-L111 

I would just remove it, but let me know what you think

**Review by aliabd — COMMENTED** [review]
Everything works as expected aside from the indentation issue

File: js/app/src/lite/custom-element/index.ts (hunk 1 above)
Comment (whitphx) [reply js/app/src/lite/custom-element/index.ts:163]: Sorry I just added it for defensively cleaning up the code without being aware of the side effect.
Just remove it.

**Review by aliabd — APPROVED** [review]
Great thanks @whitphx everything looks good now!

**Pull request merged**